Framework/Algorithms/inc/MantidAlgorithms/InterpolatingRebin.h

```diff
@@ -93,15 +96,20 @@ class DLLExport InterpolatingRebin : public Rebin {
   void outputYandEValues(API::MatrixWorkspace_const_sptr inputW,
                          const HistogramData::BinEdges &XValues_new,
                          API::MatrixWorkspace_sptr outputW);
-  void cubicInterpolation(const HistogramData::BinEdges &xOld,
-                          const MantidVec &yOld, const MantidVec &eOld,
-                          const HistogramData::BinEdges &xNew, MantidVec &yNew,
-                          MantidVec &eNew) const;
+  void
+  InterpolatingRebin::cubicInterpolation(const HistogramData::BinEdges &xOld,
+                                         const HistogramData::HistogramY &yOld,
+                                         const HistogramData::HistogramE &eOld,
+                                         const HistogramData::BinEdges &xNew,
+                                         HistogramData::HistogramY &yNew,
+                                         HistogramData::HistogramE &eNew) const;
   void noInterpolation(const HistogramData::BinEdges &xOld, const double yOld,
-                       const MantidVec &eOld,
-                       const HistogramData::BinEdges &xNew, MantidVec &yNew,
-                       MantidVec &eNew) const;
-  double estimateError(const MantidVec &xsOld, const MantidVec &esOld,
+                       const HistogramData::HistogramE &eOld,
+                       const HistogramData::BinEdges &xNew,
+                       HistogramData::HistogramY &yNew,
+                       HistogramData::HistogramE &eNew) const;
+  double estimateError(const HistogramData::HistogramX &xsOld,
```

Comment:
@DimitarTasev No, for now `HistogramData` only contains basics. I think Lamar wanted to start a list of things we might consider adding, maybe append this to that list (but in any case that would not be implemented as part of this pull request).